### PR TITLE
Add custom msbuild workspace.

### LINKED ISSUE #2: Generation fails code in a net47 project that references a netstandard2.0
The source generation fails code in a net47 project that references a netstandard2.0.

A possible workaround is to include a net47 target framework in the referenced project.

See this roslyn issue: https://github.com/dotnet/roslyn/issues/23114

### LINKED ISSUE #79: Add support for persistent Generation Host
## I'm submitting a...
- Feature request 

## Current behavior
The new Hosted Generation mode that uses out-of-process generation re-parses projects on each build, which can take a significant amont of build time.

It is important that a persistent process, such as "VBCSCompiler.exe" can stay in memory for as long as the msbuild processes that use it are running, so the parsed projects are kept in memory. 

This feature is already implemented with the in-process model, via the reuse of AppDomains.

### PR DESCRIPTION
**Pull request by jeromelaban**
GitHub Issue (If applicable): Fixes #2, fixes #79

## PR Type
What kind of change does this PR introduce?
- Performance
- Bugfix

## What is the current behavior?
- Roslyn's MSBuildWorkspace does not provided access to the inner msbuild object it creates to build the `Compilation`. The source generation tasks need it to provide to the generators, forcing the project to be parsed twice. 
- The workspace also parses all the dependent projects when a project is loaded through `MSBuildWorkspace.OpenProjectAsync`, something that the source generators don't need, and which creates #1 where projects need to have an unnecessary list of TargetFrameworks.
- The memory consumption of the generation host is too high

## What is the new behavior?
- A custom workspace is now created to reuse the msbuild instances that the source generation task already creates, building the `Compilation` object from this.
- The generation host is now using the host binary's path and the originating msbuild instance's process ID. AppDomains are now used to reuse as much as CLR loaded assemblies.
- Fixed and validated in CI the dotnet core CLI build.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tested code with current [supported SDKs](../README.md)
- [ ] Docs have been added/updated which fit [documentation template](https://github.com/nventive/Uno.SourceGeneration/tree/master/doc/.feature-template.md). (for bug fixes / features)
- [x] Tests for the changes have been added (for bug fixes / features) (if applicable)
- [x] Contains **NO** breaking changes
- [ ] Updated the [Release Notes](https://github.com/nventive/Uno.SourceGeneration/tree/master/doc/ReleaseNotes)
- [x] Associated with an issue (GitHub or internal

### CODE REVIEW

**Comment by jeromelaban** [comment]
> is this right naming?
> private static string NewMethod(SourceGeneratorContext context, string typeName, string variable)

Most probably not :) Thanks!

File: .vsts-ci.yml [hunk 1]
```diff
@@ -71,5 +90,12 @@ jobs:
       versionSpec: 4.7.0
       checkLatest: false
 
-  - bash: msbuild /r /p:Configuration=Release /p:AddSyntheticProjectReferencesForSolutionDependencies=false ./src/Uno.SourceGenerator.Linux.sln
+  - bash: msbuild /r /p:Configuration=Release ./src/Uno.SourceGenerator.Linux.sln # /bl:$(build.artifactstagingdirectory)/macos-build.binlog
```
Comment (carldebilly) [inline .vsts-ci.yml]: commented code?

**Comment by carldebilly** [comment]
TL;JA

File: .vsts-ci.yml (hunk 1 above)
Comment (jeromelaban) [reply .vsts-ci.yml]: It's often required to add it back for troubleshooting, it's easier to leave it here :)

**Pull request merged**